tasks/sealsupra/supra_config.go

```diff
@@ -0,0 +1,297 @@
+
+	requiredThreads := batchSize / sectorsPerThread
+	requiredCCX := (batchSize + sectorsPerCCX - 1) / sectorsPerCCX
+	requiredCores := requiredCCX + requiredThreads/info.ThreadsPerCore
+
+	if requiredCores > info.CoreCount {
+		return config, fmt.Errorf("not enough cores available for hashers")
+	}
+
+	coresLeftover := info.CoreCount - requiredCores
+
+	const minOverheadCores = 4
+	if coresLeftover < minOverheadCores {
+		return config, fmt.Errorf("not enough cores available for overhead")
+	}
+
+	nextFreeCore := minOverheadCores
+
+	// Assign cores for PC2 processes
+	if coresLeftover > nextFreeCore {
+		config.Topology.PC2Writer = nextFreeCore
+		nextFreeCore++
+	}
+
+	if coresLeftover > nextFreeCore {
+		config.Topology.PC2Hasher = nextFreeCore
+		nextFreeCore++
+	}
+
+	if coresLeftover > nextFreeCore {
+		config.Topology.PC2HasherCPU = nextFreeCore
+		nextFreeCore++
+	}
+
+	if coresLeftover > nextFreeCore {
+		config.Topology.PC2Reader = nextFreeCore
+		config.Topology.C1Reader = nextFreeCore
+		nextFreeCore++
+	}
+
+	// Add P2 writer cores, up to 8 total
+	if coresLeftover > nextFreeCore {
+		config.Topology.PC2Writer, config.Topology.PC2Reader = config.Topology.PC2Reader, config.Topology.PC2Writer
+
+		for i := 0; i < 7 && coresLeftover > nextFreeCore; i++ {
+			config.Topology.PC2WriterCores++
+			nextFreeCore++
+		}
+	}
+
+	sectorConfig := SectorConfig{
+		Sectors:      batchSize,
+		Coordinators: []CoordinatorConfig{},
+	}
+
+	ccxCores := make([]int, 0)
+	for i := 0; i < info.CoreCount; i += info.CoresPerL3 {
+		ccxCores = append(ccxCores, i)
+	}
+
+	for i := requiredCores; i > 0; {
+		firstCCXCoreNum := ccxCores[len(ccxCores)-1]
+		toAssign := min(i, info.CoresPerL3)
+
+		coreNum := firstCCXCoreNum + info.CoresPerL3 - toAssign
+
+		sectorConfig.Coordinators = append(sectorConfig.Coordinators, CoordinatorConfig{
+			Core:    coreNum,
+			Hashers: (toAssign - 1) * info.ThreadsPerCore,
+		})
+
+		i -= toAssign
+		if toAssign == info.CoresPerL3 {
+			ccxCores = ccxCores[:len(ccxCores)-1]
+			if len(ccxCores) == 0 {
+				break
+			}
+		}
+	}
+
+	// Reverse the order of coordinators
+	for i, j := 0, len(sectorConfig.Coordinators)-1; i < j; i, j = i+1, j-1 {
+		sectorConfig.Coordinators[i], sectorConfig.Coordinators[j] = sectorConfig.Coordinators[j], sectorConfig.Coordinators[i]
+	}
+
+	config.Topology.SectorConfigs = append(config.Topology.SectorConfigs, sectorConfig)
+
+	return config, nil
+}
+
+func FormatSupraSealConfig(config SupraSealConfig) string {
+	var sb strings.Builder
+
+	sb.WriteString("# Configuration for supra_seal\n")
+	sb.WriteString("spdk: {\n")
+	sb.WriteString("  # PCIe identifiers of NVMe drives to use to store layers\n")
+	sb.WriteString("  nvme = [ \n")
+	for i, device := range config.NVMeDevices {
+		sb.WriteString(fmt.Sprintf("           \"%s\"", device))
+
+		// comma if not last
+		if i < len(config.NVMeDevices)-1 {
+			sb.WriteString(",\n")
+		} else {
+			sb.WriteString("\n")
+		}
+	}
```

Comment:
```suggestion
w(strings.Join(lo.Map(config.NVMeDevices, func(d, i int) string { return `"`+d.String +`"`}, ",\n"))
```

Reply:
Many review matters seem unchanged. 
Have you not pushed or are you skipping this?

Reply:
That was pushed, but missed the loop below